packages/language-server/src/documentation.ts

```diff
@@ -108,3 +113,30 @@ export function getNodeDetail(node: BaseUI5Node): string {
       return node.name;
   }
 }
+
+export async function getUI5NodeName(
+  offsetAt: number,
+  text: string,
+  model?: UI5SemanticModel,
+  cachePath?: string,
+  framework?: string,
+  ui5Version?: string
+): Promise<BaseUI5Node | undefined> {
+  const documentText = text;
+  const { cst, tokenVector } = parse(documentText);
+  const ast = buildAst(cst as DocumentCstNode, tokenVector);
+  const offset = offsetAt;
+  const astPosition = astPositionAtOffset(ast, offset);
+  if (!model) {
+    model = await fetchModel(cachePath, framework, ui5Version);
+  }
+  if (astPosition !== undefined) {
+    return findUI5HoverNodeAtOffset(astPosition, model);
+  }
+  return undefined;
+}
+
+async function fetchModel(cachePath, framework, ui5Version) {
+  const ui5Model = await getSemanticModel(cachePath, framework, ui5Version);
```

Comment:
I do not recall:
Does every functional flow invoke `getSemanticModel` ?
Or does it happen once on language server startup?

Is this different or the same as other functional flows?